### PR TITLE
Improveed javascript code docs

### PR DESCRIPTION
**Pull request by 5p0rt5BEArD**
Specific examples for use of GameState
Explained ds.encode with example
Linking to ds-examples repo (Leaning on this a lot!)

### CODE REVIEW

File: docs/code-docs/plugin-javascript.md [hunk 1]
```diff
@@ -4,23 +4,69 @@
...
-
-## Help inspecting the shape of data
-
-While playing, hit F12 and turn on verbose console logging. Each time you select a building on the map you will get an output of all the state that is being sent to that buildings javascript plugin. Expanding the elements of these data objects is a good way to see what data is available when writing your own plugin.
+The plugin's `Update()` function is passed a parameter of type [`GameSate`](https://github.com/playmint/ds/blob/4b541a8e4c05d84a392c0cffec850d8b6949cd96/core/src/types.ts#L330) that you can use to check who (or what?) is using the building and configure your UI accordingly. 
+
+GameState is split into three top level objects: `player`, `world` and `selected`.
+
+
+These docs are not a complete and up to dat reference so the best way to see th state available is to use the **browser debug console**:
```
Comment (chrisfarms) [inline docs/code-docs/plugin-javascript.md:18]:
```suggestion
These docs are not a complete and up to date reference so the best way to see th state available is to use the **browser debug console**:
```

File: docs/code-docs/plugin-javascript.md [hunk 2]
```diff
@@ -4,23 +4,69 @@
...
-
-************World************ contains summary information about everything in the Downstream world including ids for constructed buildings and discovered tiles. Not all of the world state is available but all state related to currently selected tiles is available from the Selection object.
-
-********************Selection******************** contains all state related to the currently selected MobileUnit, currently selected tile(s) and the intent (one of the Select, Construct, Scout, Move and Use modes) of the player.
-
-## Help inspecting the shape of data
-
-While playing, hit F12 and turn on verbose console logging. Each time you select a building on the map you will get an output of all the state that is being sent to that buildings javascript plugin. Expanding the elements of these data objects is a good way to see what data is available when writing your own plugin.
+The plugin's `Update()` function is passed a parameter of type [`GameSate`](https://github.com/playmint/ds/blob/4b541a8e4c05d84a392c0cffec850d8b6949cd96/core/src/types.ts#L330) that you can use to check who (or what?) is using the building and configure your UI accordingly.
```
Comment (chrisfarms) [inline docs/code-docs/plugin-javascript.md:13]:
```suggestion
The plugin's `Update()` function is passed a parameter of type [`GameSate`](https://github.com/playmint/ds/blob/4b541a8e4c05d84a392c0cffec850d8b6949cd96/core/src/types.ts#L330) that you can use to inspect the current state of the game, discover what the player has selected and have your UI react accordingly. 
```

File: docs/code-docs/plugin-javascript.md [hunk 3]
```diff
@@ -4,23 +4,69 @@
...
+    > ```javascript
+    > const mobileUnit = selected.mobileUnit
+    > ```
+- The top bag belonging to the MobileUnit. 
+    - *bags are an array of equpment objects where index 0 is bot guranteed to have key 0*
+    - *the actual bag is a subobject of the bag (!)(unless we've renamed this nonsense by the time you read this)*
+    > ```javascript
+    > const topBag = mobileUnit.bags.find(b => b.key == 0).bag;
+    > ```
+- Find out if a bag contains a particulatr item.
```
Comment (chrisfarms) [inline docs/code-docs/plugin-javascript.md:49]:
```suggestion
- Find out if a bag contains a particular item. 
```

File: docs/code-docs/plugin-javascript.md [hunk 4]
```diff
@@ -163,8 +209,15 @@ ds.dispatch(
...
-ds.encodeCall();
+// A soldidty function in your own building contract can be called by encoding its name and call parameters
+// and passing it as the payload of the use building function.
+// To do this the call needs to be abi encoded for decoding on the solidty side
+// See the ds-examples repo and the PostOffice building for an example of this in action
+const payload = ds.encodeCall("function panic()", []);
+ds.dispatch({
+      name: "BUILDING_USE",
+      args: [selected.tiles[0].building.id, selected.mobileUnit.id, payload],
+    });
```
Comment (chrisfarms) [inline docs/code-docs/plugin-javascript.md:220]: The way this is worded makes it sound like you can execute a remote function, which you can not. Maybe something like this is more clear:

```suggestion
// helper to abi encode values for a payload that you can abi.decode in your contract code
// use this to pass arbitrary parameters to your BuildingKind contract's use() function.
// See the ds-examples repo and the PostOffice building for an example of this in action
const payload = ds.encodeCall("function values(uint256 aNumber, string memory aString)", [1, "example"]);
ds.dispatch({
      name: "BUILDING_USE",
      args: [selected.tiles[0].building.id, selected.mobileUnit.id, payload],
    });
```

**Review by chrisfarms — APPROVED** [review]
some suggestions but I'm not great with the words :)

**Comment by 5p0rt5BEArD** [comment]
Thanks for the edits @chrisfarms 🙏

**Pull request merged**